plane2/src/typed_socket/mod.rs

```diff
@@ -2,19 +2,61 @@ use crate::PlaneVersionInfo;
 use serde::de::DeserializeOwned;
 use serde::{Deserialize, Serialize};
 use std::fmt::Debug;
+use tokio::sync::mpsc::{Receiver, Sender};
 
 pub mod client;
 pub mod server;
 
+pub enum SocketAction<T> {
+    Send(T),
+    Close,
+}
+
 pub trait ChannelMessage: Send + Sync + 'static + DeserializeOwned + Serialize + Debug {
     type Reply: ChannelMessage<Reply = Self>;
 }
 
-#[async_trait::async_trait]
-pub trait FullDuplexChannel<T: ChannelMessage> {
-    async fn send(&mut self, message: &T) -> anyhow::Result<()>;
+pub struct TypedSocket<T: ChannelMessage> {
```

Comment:
`TypedSocket` now represents the handle to a connection whether it is a client or server connection. It replaces both the old `TypedSocket` and `TypedSocketServer`, and obviates the need for the `FullDuplexChannel` trait which they both inhereted.